### PR TITLE
bumpup version for rocm-debug-agent,rocprofiler,rocsolver for rocm-4.…

### PR DESCRIPTION
**Pull request by srekolam**
…1.0 rel

### CODE REVIEW

File: var/spack/repos/builtin/packages/rocsolver/package.py [hunk 1]
```diff
@@ -26,21 +27,25 @@ class Rocsolver(CMakePackage):
...
         depends_on('rocblas@' + ver, type='link', when='@' + ver)
 
     def cmake_args(self):
+        incl = self.spec['rocblas'].prefix
         args = [
             '-DBUILD_CLIENTS_SAMPLES=OFF',
             '-DBUILD_CLIENTS_TESTS=OFF',
             '-DBUILD_CLIENTS_BENCHMARKS=OFF'
         ]
+        if self.spec.satisfies('@4.1.0:'):
```
Comment (haampie) [inline var/spack/repos/builtin/packages/rocsolver/package.py:46]: Why is this necessary? It shouldn't be?

Comment (srekolam) [reply var/spack/repos/builtin/packages/rocsolver/package.py:46]: I have included as for rocm-4.1.0, in compiling this file , /tmp/root/spack-stage/spack-stage-rocsolver-4.1.0-ifnhpjic5mvqbph7y7wz3utudmoaiace/spack-src/rocsolver/library/src/auxiliary/rocauxiliary_orgtr_ungtr.cpp , i see the below error ...

In file included from /tmp/root/spack-stage/spack-stage-rocsolver-4.1.0-ifnhpjic5mvqbph7y7wz3utudmoaiace/spack-src/rocsolver/library/src/buildinfo.cpp:5:
In file included from /tmp/root/spack-stage/spack-stage-rocsolver-4.1.0-ifnhpjic5mvqbph7y7wz3utudmoaiace/spack-src/rocsolver/library/src/include/rocblas.hpp:10:
In file included from /tmp/root/spack-stage/spack-stage-rocsolver-4.1.0-ifnhpjic5mvqbph7y7wz3utudmoaiace/spack-src/rocsolver/library/src/include/common_device_helpers.hpp:8:
In file included from /tmp/root/spack-stage/spack-stage-rocsolver-4.1.0-ifnhpjic5mvqbph7y7wz3utudmoaiace/spack-src/rocsolver/library/src/include/libcommon.hpp:7:
/tmp/root/spack-stage/spack-stage-rocsolver-4.1.0-ifnhpjic5mvqbph7y7wz3utudmoaiace/spack-src/rocsolver/library/src/include/../../../../rocblascommon/library/src/include/utility.h:8:10: fatal error: 'rocblas.h' file not found #include "rocblas.h"

Comment (haampie) [reply var/spack/repos/builtin/packages/rocsolver/package.py:46]: did you make an issue for this?

Comment (srekolam) [reply var/spack/repos/builtin/packages/rocsolver/package.py:46]: [rocsolver_4.1.0_build_error.txt](https://github.com/spack/spack/files/6244034/rocsolver_4.1.0_build_error.txt)
No, Not yet.. I have attached the full error text for your reference..

Comment (haampie) [reply var/spack/repos/builtin/packages/rocsolver/package.py:46]: okay, that's weird, i don't see rocblas among the include flags anywhere... might be worth looking at the CMakeCache.txt file (e.g. use `spack install --keep-stage rocsolver` or `spack stage rocsolver` `spack cd --stage rocsolver` `spack build-env rocsolver -- /bin/bash` `mkdir build` `cd build` `cmake ..`)

Comment (srekolam) [reply var/spack/repos/builtin/packages/rocsolver/package.py:46]: [rocsolver_4.1.0_CMakeCache.txt](https://github.com/spack/spack/files/6245899/rocsolver_4.1.0_CMakeCache.txt)
Attached is the CMakeCache.txt with out the -DCMAKE_CXX_FLAGS:String=-I{0}/rocblas/include'.format(incl) change.

**Review by tldahlgren — APPROVED** [review]
Confirmed new versions available in each repository

**Comment by srekolam** [comment]
@haampie  , can you please let me know if this PR is okay.

**Comment by haampie** [comment]
Let's first get #22762 in and see if the rest of the packages still work after that.

**Review by haampie — APPROVED** [review]
